internal/payloads/reservation_payload.go

```diff
@@ -215,7 +218,10 @@ type GCPReservationRequestPayload struct {
 	SourceID string `json:"source_id" yaml:"source_id"`
 
 	// Optional launch template name global/instanceTemplates/NAME or empty string
-	LaunchTemplateName string `json:"launch_template_name,omitempty"`
+	LaunchTemplateName string `json:"launch_template_name,omitempty" yaml:"launch_template_name"`
+
+	// Optional name pattern of the instance(s).
+	Name string `json:"name_pattern" yaml:"name_pattern"`
```

Comment:
This should be refactored, creating different JSON names and Go field type names is something we should definitely avoid.

```suggestion
	NamePattern string `json:"name_pattern" yaml:"name_pattern"`
```

It is fine to keep this as `Name` in the response as long as the JSON/YAML has the same name. But for request, I think it is more appropriate to name it as a pattern.

Reply:
So regarding the Name pattern, in a bulkInsert Request, we do not have a name field but a name pattern field which needs to have the ### sign. 
I will change the GCP naming from Name to NamePattern

Reply:
Yeah I get that, it is okay to have Name in response but NamePattern in request. It seems logical and correct to me :-)